src/app/services/services.module.ts

```diff
@@ -48,107 +40,52 @@ import { WebPlatformUtilsService } from "../../services/webPlatformUtils.service
 import { PolicyListService } from "./policy-list.service";
 import { RouterService } from "./router.service";
 
-export function initFactory(
-  window: Window,
-  environmentService: EnvironmentServiceAbstraction,
-  notificationsService: NotificationsServiceAbstraction,
-  vaultTimeoutService: VaultTimeoutService,
-  i18nService: I18nService,
-  eventLoggingService: EventLoggingService,
-  twoFactorService: TwoFactorServiceAbstraction,
-  stateService: StateServiceAbstraction,
-  platformUtilsService: PlatformUtilsServiceAbstraction,
-  cryptoService: CryptoServiceAbstraction
-): () => void {
-  return async () => {
-    await stateService.init();
-
-    const urls = process.env.URLS as Urls;
-    urls.base ??= window.location.origin;
-    environmentService.setUrls(urls);
-
-    setTimeout(() => notificationsService.init(), 3000);
-
-    vaultTimeoutService.init(true);
-    const locale = await stateService.getLocale();
-    await i18nService.init(locale);
-    eventLoggingService.init(true);
-    twoFactorService.init();
-    const htmlEl = window.document.documentElement;
-    htmlEl.classList.add("locale_" + i18nService.translationLocale);
-
-    // Initial theme is set in index.html which must be updated if there are any changes to theming logic
-    platformUtilsService.onDefaultSystemThemeChange(async (sysTheme) => {
-      const bwTheme = await stateService.getTheme();
-      if (bwTheme === ThemeType.System) {
-        htmlEl.classList.remove("theme_" + ThemeType.Light, "theme_" + ThemeType.Dark);
-        htmlEl.classList.add("theme_" + sysTheme);
-      }
-    });
-
-    const containerService = new ContainerService(cryptoService);
-    containerService.attachToWindow(window);
-  };
-}
-
 @NgModule({
   imports: [ToastrModule, JslibServicesModule],
   declarations: [],
   providers: [
-    {
-      provide: APP_INITIALIZER,
-      useFactory: initFactory,
-      deps: [
-        "WINDOW",
-        EnvironmentServiceAbstraction,
-        NotificationsServiceAbstraction,
-        VaultTimeoutServiceAbstraction,
-        I18nServiceAbstraction,
-        EventLoggingServiceAbstraction,
-        TwoFactorServiceAbstraction,
-        StateServiceAbstraction,
-        PlatformUtilsServiceAbstraction,
-        CryptoServiceAbstraction,
-      ],
-      multi: true,
-    },
+    InitService,
     OrganizationGuardService,
     OrganizationTypeGuardService,
     RouterService,
     EventService,
     PolicyListService,
+    {
+      provide: APP_INITIALIZER,
+      useFactory: (initService: InitService) => initService.init(),
+      deps: [InitService],
+      multi: true,
+    },
+    {
+      provide: STATE_FACTORY,
+      useValue: new StateFactory(GlobalState, Account),
+    },
+    {
+      provide: STATE_SERVICE_USE_CACHE,
+      useValue: false,
+    },
     {
       provide: I18nServiceAbstraction,
       useFactory: (window: Window) => new I18nService(window.navigator.language, "locales"),
```

Comment:
Thoughts on moving `"locales"` to a variable?

Reply:
I assume you mean token - done, depends on https://github.com/bitwarden/jslib/pull/783